pkg/cmd/migrate.go

```diff
@@ -0,0 +1,88 @@
+package cmd
+
+import (
+	"database/sql"
+	"fmt"
+
+	_ "github.com/jackc/pgx/v4/stdlib"
+	"github.com/openfga/openfga/assets"
+	"github.com/pressly/goose/v3"
+	"github.com/spf13/cobra"
+)
+
+const (
+	datastoreEngineFlag = "datastore-engine"
+	datastoreURIFlag    = "datastore-uri"
+	versionFlag         = "version"
+)
+
+func NewMigrateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "migrate",
+		Short: "Run database schema migrations needed for the OpenFGA server",
+		Long:  `The migrate command is used to migrate the database schema needed for OpenFGA.`,
+		RunE:  runMigration,
+	}
+
+	cmd.Flags().String(datastoreEngineFlag, "", "(required) the database engine to run the migrations for")
+	if err := cmd.MarkFlagRequired(datastoreEngineFlag); err != nil {
+		panic(err)
+	}
+
+	cmd.Flags().String(datastoreURIFlag, "", "(required) the connection uri of the database to run the migrations against (e.g. 'postgres://postgres:password@localhost:5432/postgres')")
+	if err := cmd.MarkFlagRequired(datastoreURIFlag); err != nil {
```

Comment:
Can we add this panic functionality to `cmdutil` (pkg/cmd/util). For example,

```
// MustMarkFlagRequired attempts to mark the provided flag key as required and panics
// if it fails to do so.
func MustMarkFlagRequired(key string, cmd) {
    if err := cmd.MarkFlagRequired(key); err != nil {
        panic(err)
    }
}
```

or something like that..

Reply:
Function added. Thanks!